tests/unit/lib/SGSparseMatrix_unittest.cc

```diff
@@ -184,37 +184,44 @@ TEST(SGSparseMatrix, io_libsvm)
 
 TEST(SGSparseMatrix, access_by_index_non_square)
 {
-    //GeneratSparseMatrix function cannot used in this test since it uses () operator
-    CRandom randGenerator(0);
-    float64_t sparseLevel=0.1;
+    const float64_t sparseLevel=0.1;
     const index_t numberOfFeatures=50;
     const index_t numberOfVectors=100;
-
-    SGSparseMatrix<float64_t> matrix(numberOfFeatures, numberOfVectors);
-
+    const index_t randSeed=0;
+
+    //generate a sparse and dense matrix and compare this entries
+    SGSparseMatrix<float64_t> sparseMatrix(numberOfFeatures, numberOfVectors);
+    GenerateMatrixAccrodingToSparseConvention<SGSparseMatrix<float64_t> >(sparseLevel,
+                                                                          numberOfFeatures,
+                                                                          numberOfVectors,
+                                                                          randSeed,
+                                                                          &sparseMatrix);
```

Comment:
I am not really a fan of this sort of indentation, but if you like it I guess it's fine.